purchases/src/main/kotlin/com/revenuecat/purchases/CustomerInfoUpdater.kt

```diff
@@ -0,0 +1,82 @@
+package com.revenuecat.purchases
+
+import android.os.Handler
+import android.os.Looper
+import com.revenuecat.purchases.common.LogIntent
+import com.revenuecat.purchases.common.caching.DeviceCache
+import com.revenuecat.purchases.common.log
+import com.revenuecat.purchases.common.offlineentitlements.OfflineEntitlementsManager
+import com.revenuecat.purchases.identity.IdentityManager
+import com.revenuecat.purchases.interfaces.UpdatedCustomerInfoListener
+import com.revenuecat.purchases.strings.ConfigureStrings
+import com.revenuecat.purchases.strings.CustomerInfoStrings
+
+/**
+ * This class is responsible for updating the customer info cache and notifying the listeners.
+ */
+internal class CustomerInfoUpdater(
```

Comment:
Honestly I'm not sure if this was the best name but I thought it was ok... But I explained a bit in the comment for this class.

Reply:
I think it's fine, but if requires explanation maybe it's worth giving it another thought. I don't have a better idea, but it's true that updater makes it seem like it will fetch it

Reply:
I renamed it to `CustomerInfoUpdateReceiver`, what to you think?

Reply:
I am not convinced about the Receiver part, since it's also notifying. What about `CustomerInfoUpdatesHandler`?

Reply:
I like that, I rename to the singular version of it `CustomerInfoUpdateHandler` in https://github.com/RevenueCat/purchases-android/pull/1052/commits/e9d41c1544c9d7994fb3f1e0cd8422ce8cd5be7c